src/Compilers/CSharp/Portable/Lowering/LocalRewriter/LocalRewriter_PatternSwitchStatement.cs

```diff
@@ -60,13 +60,20 @@ private BoundStatement MakeLoweredForm(BoundPatternSwitchStatement node)
                 var expression = _localRewriter.VisitExpression(node.Expression);
                 var result = ArrayBuilder<BoundStatement>.GetInstance();
 
-                if (expression.ConstantValue == null)
+                // EnC: We need to insert a hidden sequence point to handle function remapping in case
+                // the containing method is edited while methods invoked in the expression are being executed.
+                if (!node.WasCompilerGenerated && _localRewriter.Instrument)
                 {
-                    // EnC: We need to insert a hidden sequence point to handle function remapping in case
-                    // the containing method is edited while methods invoked in the expression are being executed.
-                    if (!node.WasCompilerGenerated && _localRewriter.Instrument)
+                    var instrumentedExpression = _localRewriter._instrumenter.InstrumentSwitchStatementExpression(node, expression, _factory);
+                    if (expression.ConstantValue == null)
                     {
-                        expression = _localRewriter._instrumenter.InstrumentSwitchStatementExpression(node, expression, _factory);
+                        expression = instrumentedExpression;
+                    }
+                    else
+                    {
+                        // If the expression is a constant, we leave it alone (the decision tree lowering code needs
+                        // to see that constant). But we add an additional leading statement with the instrumented expression.
+                        result.Add(_factory.ExpressionStatement(instrumentedExpression));
```

Comment:
In the case of a constant switch expression, no hidden sequence point is inserted, but instead an extra copy is inserted by the debug instrumentation code.